### PR TITLE
Add model and handler for duplicates, update admin export

### PR DESCRIPTION
**Pull request by pjsier**
Should implement most of what's mentioned in #236, other than the admin interface. I created a `RequestDuplicate` model that just has the default fields and `requestId` as an association with the `Requests` table. Wanted to get some feedback before moving to the admin side of things though.

I tried to handle the promises as cleanly as possible in the `saveOrDuplicateRequest` function, but I'm not sure if there was a way of avoiding the nested promises there altogether. The `RequestDuplicate` model file was created from `sequelize-cli`, so to manage the association on the `Request` side of things I updated the request model file with the same general structure.

### CODE REVIEW

File: models/request.js [hunk 1]
```diff
@@ -17,20 +17,27 @@
...
+    city: DataTypes.TEXT,
+    state: DataTypes.TEXT,
+    zip: DataTypes.TEXT,
+          sms_raw_zip: DataTypes.TEXT,
+    phone: DataTypes.TEXT,
+          sms_raw_phone: DataTypes.TEXT,
+    email: DataTypes.TEXT,
+          source: DataTypes.TEXT,
+          serial: { type: DataTypes.TEXT, unique: true },
+          status: DataTypes.TEXT
```
Comment (cecilia-donnelly) [inline models/request.js]: Try to minimize these unrelated whitespace changes as much as possible.  It makes the diff confusing to read.  And I know the indentation wasn't consistent to begin with, but now it's even less so.  Can you consistify, please?

File: models/requestduplicate.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/* Smoke Alarm Installation Request Portal (getasmokealarm.org)
+ * 
+ * Copyright (C) 2015  American Red Cross
```
Comment (cecilia-donnelly) [inline models/requestduplicate.js]: Also 2017.  (I assume these are just copy/pasto's, no problem.)

File: views/utilities.js [hunk 3]
```diff
@@ -227,6 +227,39 @@ module.exports  = {
...
     },
 
+    // Searches for any previous request to see if this is a duplicate
+    // Returns null if not found, otherwise returns first record
+    findPriorRequest: function(requestData) {
+      // Replacing non-numeric characters with Postgres match operator, adding 
+      // one to the beginning to accommodate country codes
+      var phoneQuery = "%" + requestData.phone.replace(/[^0-9]/g, "%");
+
+      return db.Request.findOne({
```
Comment (cecilia-donnelly) [inline views/utilities.js]: Should this be just a regular `find`?  What if there's more than one duplicate?  (I don't think I mentioned this case in the issue, but...maybe we take the most recent duplicate?)

File: views/utilities.js [hunk 4]
```diff
@@ -227,6 +227,39 @@ module.exports  = {
...
+          }
+      });
+    },
+
+    // Force update of updatedAt without changing attributes
+    // source: https://github.com/sequelize/sequelize/issues/3759
+    updateRequestTime: function(request) {
+      request.changed('updatedAt', true);
+      return request.save();
+    },
```
Comment (cecilia-donnelly) [inline views/utilities.js]: This is cool!

Comment (cecilia-donnelly) [reply views/utilities.js]: I guess the only concern would be if it might not continue to work through later versions of Sequelize?  Any thoughts about that?

File: models/requestduplicate.js (hunk 2 above)
Comment (pjsier) [reply models/requestduplicate.js]: Yep, I'd just copied and pasted these, thanks for catching it!

File: views/utilities.js (hunk 3 above)
Comment (pjsier) [reply views/utilities.js]: I think it'd have to be the most recent, even if there are more than one, so I added an ordering by `createdAt` in the latest commit

File: views/utilities.js (hunk 4 above)
Comment (pjsier) [reply views/utilities.js]: Hmm, from the PR comments it looks like they may add a better way of doing this, but that this is using internal methods that don't look like they'll be deprecated. The only other way I've seen is to update a property that doesn't actually exist on the instance

File: models/request.js (hunk 1 above)
Comment (pjsier) [reply models/request.js]: Sorry about that! Part of the indentation change was because of the different structure for the models based off of Sequelize, but the indentation definitely needed cleaning. Fixed in the latest commit

File: views/utilities.js (hunk 3 above)
Comment (cecilia-donnelly) [reply views/utilities.js]: +1

**Comment by cecilia-donnelly** [comment]
Great!  As far as I'm concerned, go ahead and add the admin side of this.  Hop into channel if you want to discuss anything.

**Comment by cecilia-donnelly** [comment]
@pjsier, you wrote: "is the request detail view on admin currently working? want to make sure I didn't break it, but I don't see any way to get to it through the UI, and it doesn't work when I enter the URL."

No, it's not working currently.

**Comment by pjsier** [comment]
Updated with the changes mentioned in #236. I spent too long trying to count the duplicate requests in SQL, but given there aren't too many it ended up being a lot more explicit to just count them afterwards, and for `json2csv` to work the objects needed to be converted to JSON anyway.

File: views/utilities.js [hunk 5]
```diff
@@ -227,6 +227,40 @@ module.exports  = {
...
+    // Returns null if not found, otherwise returns first record
+    findPriorRequest: function(requestData) {
+      // Replacing non-numeric characters with Postgres match operator, adding 
+      // one to the beginning to accommodate country codes
+      var phoneQuery = "%" + requestData.phone.replace(/[^0-9]/g, "%");
+
+      return db.Request.findOne({
+          where: {
+            name: { $ilike: requestData.name },
+            source: requestData.source,
```
Comment (cecilia-donnelly) [inline views/utilities.js]: I actually think that even if the source is different we should still mark it as a duplicate!  I can see a case where someone called 311 and then also made a request by text message with the same information -- in that case I think it should still be marked as a duplicate.  What do you think?

Comment (pjsier) [reply views/utilities.js]: I'd agree with that, good point! I'll make that update

**Comment by pjsier** [comment]
@cecilia-donnelly I think it should just be `sequelize db:migrate` to create the table (no seed data needed), but I don't think that's mentioned in the current INSTALL.md. I'm going to do a test run from scratch to make sure that works, and then is INSTALL.md the only place to update those instructions?
